snapshot/snapshot.go

```diff
@@ -146,8 +162,10 @@ func NewSnapshotter(ctx context.Context, root string, targetFs FileSystem, opts
 		userxattr:   userxattr,
 	}
 
-	if err := o.restoreRemoteSnapshot(ctx); err != nil {
-		return nil, errors.Wrap(err, "failed to restore remote snapshot")
+	if config.restoreSnapshots {
+		if err := o.restoreRemoteSnapshot(ctx); err != nil {
+			return nil, errors.Wrap(err, "failed to restore remote snapshot")
+		}
```

Comment:
We should not change the default behaviour here. Restoring should be enabled by default and disabled by an option.